x-pack/test_serverless/api_integration/services/alerting_api.ts

```diff
@@ -12,33 +12,43 @@ import type {
 
 import { MetricThresholdParams } from '@kbn/infra-plugin/common/alerting/metrics';
 import { ThresholdParams } from '@kbn/observability-plugin/common/custom_threshold_rule/types';
+import {
+  RoleCredentials,
+  InternalRequestHeader,
+  SupertestWithoutAuthType,
+} from '../../shared/services';
 import { SloBurnRateRuleParams } from './slo_api';
 import { FtrProviderContext } from '../ftr_provider_context';
 
 export function AlertingApiProvider({ getService }: FtrProviderContext) {
   const retry = getService('retry');
-  const supertest = getService('supertest');
   const es = getService('es');
   const requestTimeout = 30 * 1000;
   const retryTimeout = 120 * 1000;
   const logger = getService('log');
 
   return {
     async waitForRuleStatus({
+      supertestWithoutAuth,
+      roleAuthc,
+      internalReqHeader,
```

Comment:
But the api key header has to be re-made before every call, then must be invalidated in the after method.
I could see a HOF that closes over a newly generated api key and returns a function that invalidates the api key.  But I'm just spitballing.  It seems like a leaky contract to me, especially since the invalidation HAS TO HAPPEN.

But, I like being flexible.  I'm open to ideas.

Reply:
I think we could simplify things here. Since we are in a service here, we don't need to pass the supertest instance at all and could use the `supertestWithoutAuth` service directly. The same goes for the intzernal reuqest headers - no need to pass them, we can just get them directly. The only thing we really need is the `roleAuthc` to tell us which role to use for the request.